AbstractScreen.java

```diff
@@ -85,7 +85,7 @@ public int hashCode() {
 
   }
 
-  private static class TextFieldWithPrompt extends JTextField {
+  public static class TextFieldWithPrompt extends JTextField {

```

Comment:
Sorry to be nitpicky, but can you keep this private and just make the return type JTextField in the new method? It will still work because this extends JTextField.

This is more consistent with what is done with HashableButton, for instance

Reply:
got it! that's better!